frontend/src/routes/wallet/keypair/metadata.tsx

```diff
@@ -0,0 +1,313 @@
+import { useCallback, useState } from 'react'
+import type { DropResult } from 'react-beautiful-dnd'
+import { DragDropContext, Draggable, Droppable } from 'react-beautiful-dnd'
+import { useFieldArray, useForm } from 'react-hook-form'
+
+import { Button } from '../../../components/button'
+import { FormGroup } from '../../../components/form-group'
+import { Input } from '../../../components/forms/input'
+import { Header } from '../../../components/header'
+import { requestPassphrase } from '../../../components/passphrase-modal'
+import { AppToaster } from '../../../components/toaster'
+import { Colors } from '../../../config/colors'
+import { Intent } from '../../../config/intent'
+import { updateKeyPairAction } from '../../../contexts/global/global-actions'
+import type { GlobalDispatch } from '../../../contexts/global/global-context'
+import { useGlobal } from '../../../contexts/global/global-context'
+import { useCurrentKeypair } from '../../../hooks/use-current-keypair'
+import { Validation } from '../../../lib/form-validation'
+import { createLogger } from '../../../lib/logging'
+import { Service } from '../../../service'
+import type { Meta } from '../../../wailsjs/go/models'
+import { AnnotateKeyRequest } from '../../../wailsjs/go/models'
+
+const notName = (value: string) =>
+  value === 'name' ? 'Name is already in use' : true
+
+const rowStyles = {
+  width: '100%',
+  display: 'grid',
+  gridTemplateColumns: '20px 1fr 1fr auto',
+  gap: 12
+}
+
+const cellStyles = {
+  display: 'flex',
+  alignItems: 'center',
+  height: 40,
+  marginTop: 5
+}
+
+const underlined = {
+  textDecoration: 'underline'
+}
+
+const logger = createLogger('Metadata')
+
+const useMetaUpdate = (
+  dispatch: GlobalDispatch,
+  pubKey?: string,
+  wallet?: string
+) => {
+  const [loading, setLoading] = useState(false)
+
+  const update = useCallback(
+    async (metadata: Meta[]) => {
+      setLoading(true)
+      try {
+        if (!pubKey || !wallet) {
+          return
+        }
+
+        const passphrase = await requestPassphrase()
+        await Service.AnnotateKey(
+          new AnnotateKeyRequest({
+            wallet,
+            pubKey,
+            metadata,
+            passphrase
+          })
+        )
+
+        const keypair = await Service.DescribeKey({
+          wallet,
+          passphrase,
+          pubKey
+        })
+
+        dispatch(updateKeyPairAction(wallet, keypair))
+
+        AppToaster.show({
+          message: `Successfully updated metadata`,
+          intent: Intent.SUCCESS
+        })
+        setLoading(false)
```

Comment:
thought (non-blocking): I wonder if there is a useBackend hook we could have for doing all the loading/error logic that is duplicated